src/validation/statement.rs

```diff
@@ -1123,7 +1130,7 @@ fn validate_assignment<T: AnnotationMap>(
             && is_valid_assignment(left_type, right_type, right, context.index, location, validator))
         {
             // TODO: #THIS && !left_type.is_this()
-            if left_type.is_pointer() && right_type.is_pointer() {
+            if left_type.is_pointer() && left_type.is_type_safe_pointer() && right_type.is_pointer() {
```

Comment:
I think just checking for `left_type.is_type_safe_pointer()` is enough here? Again we're just matching slightly different patterns twice